### ISSUE TITLE
81dd496eb380aa051494f93fd99322ec796ec4c2 breaks stderr after exiting subshell

### ISSUE THREAD

**Issue by nathanchance** [sindresorhus/pure#550]
#### System report (output of `prompt_pure_system_report`):

```
- Zsh: zsh 5.8 (x86_64-pc-linux-gnu) (/home/nathan/usr/bin/zsh)
- Operating system: Linux (#1 SMP Sun Jun 21 12:31:14 MST 2020)
- Terminal program: unknown (unknown)
- Tmux: yes
- Git: git version 2.27.0
- Pure state:
    - username: `''`
    - prompt: `❯`
    - version: `1.11.0`
- PROMPT: `typeset -g PROMPT=$'%F{${prompt_pure_colors[path]}}%~%f %F{242}${prompt_pure_vcs_info[branch]}%F{218}${prompt_pure_git_dirty}%f
%{
%}%(12V.%F{$prompt_pure_colors[virtualenv]}%12v%f .)%(?.%F{$prompt_pure_colors[prompt:success]}.%F{$prompt_pure_colors[prompt:error]})${prompt_pure_state[prompt]}%f '`
- Colors: `typeset -g -A prompt_pure_colors=( [execution_time]=yellow [git:action]=yellow [git:arrow]=cyan [git:branch]=242 [git:branch:cached]=red [git:dirty]=218 [git:stash]=cyan [host]=242 [path]=blue [prompt:continuation]=242 [prompt:error]=red [prompt:success]=magenta [user]=242 [user:root]=default [virtualenv]=242 )`
- Virtualenv: `export VIRTUAL_ENV_DISABLE_PROMPT=12`
- Conda: `export CONDA_CHANGEPS1=no`
- Detected frameworks: None

```

#### I have:

- [ x ] Tested with another terminal program and can reproduce the issue: Windows Terminal and Blink Shell on iOS
- [ x ] Followed the [integration](https://github.com/sindresorhus/pure#integration) instructions for my framework (I have no framework).

### Problem description

81dd496eb380aa051494f93fd99322ec796ec4c2 breaks stderr after exiting a subshell. I use this all over my place in my dotfiles so I can have self contained functions and it has destroyed my workflow (as I no longer see errors about undefined functions or incorrect commands).

Before:

![image](https://user-images.githubusercontent.com/11478138/85336890-4ee4d300-b494-11ea-974f-b47feedd6d6e.png)

After:

![image](https://user-images.githubusercontent.com/11478138/85336948-6ae87480-b494-11ea-9c19-b18aa9d1dca2.png)

### Reproduction steps

```
$ echo "test" > /dev/stderr

$ (exit)

$ echo "test" > /dev/stderr
```

### My `.zshrc`:

```shell
fpath+=(~/.zfunctions)
autoload -U promptinit
promptinit
prompt pure
```

**Comment by mafredri** [comment]
Hi @nathanchance, I've been unable to reproduce this. There's been similar issues before #547 and https://github.com/mafredri/zsh-async/issues/35, but these should have been fixed in zsh-async `>= v1.8.1`. You can check which version you have via `echo $ASYNC_VERSION`.

**Comment by nathanchance** [comment]
Well that would certainly make sense since it seems that the version of `async.zsh` that is bundled in this repository is `1.8.0` :)

If I manually update `async.zsh` to the latest from your tree, the issue is resolved. Thanks for the comment, I will keep the copy updated manually until it is updated here.
